src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkInsertIndex(index);
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == EMPTY_LIST) {
+            newNode.next = first;
+            first = newNode;
+            size++;
+            return;
+        }
+        Node<T> temp = first;
+        for (int i = 0; i < index - OFFSET_ONE; i++) {
+            temp = temp.next;
+        }
+        temp.next.prev = newNode;
+        newNode.next = temp.next;
+        newNode.prev = temp;
+        temp.next = newNode;
+
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        T[] listToArray = (T[]) list.toArray();
+        for (T values : listToArray) {
+            linkLast(values);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkAccessIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkAccessIndex(index);
+        final T initialItem;
+        if (index == EMPTY_LIST) {
+            initialItem = first.item;
+            first.item = value;
+            return initialItem;
+        }
+        if (index == size - OFFSET_ONE) {
+            initialItem = last.item;
+            last.item = value;
+            return initialItem;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> temp = findNodeByIndex(index);
+
+        temp.next.prev = newNode;
+        temp.prev.next = newNode;
+        newNode.prev = temp.prev;
+        newNode.next = temp.next;
+
+        return temp.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkAccessIndex(index);
+        if (index == EMPTY_LIST) {
+            return removeFirst();
+        }
+        if (index == size - OFFSET_ONE) {
+            return removeLast();
+        }
+        Node<T> temp = findNodeByIndex(index);
+        unlink(temp);
+        return temp.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> temp = first;
+        while (temp != null) {
+            if (object != null && temp.equals(first) && object.equals(first.item)) {
+                removeFirst();
+                return true;
+            }
+            if (object != null && temp.equals(last) && object.equals(last.item)) {
+                removeLast();
+                return true;
+            }
+            if (Objects.equals(temp.item, object)) {
+                unlink(temp);
+                return true;
+            }
+            temp = temp.next;
+        }
```

Comment:
Don't iterate your nodes from the head to the tail or vice versa without considering the index. You can improve performance by iterating only half of the list depending on the index value.